server/openapi.go

```diff
@@ -89,11 +115,42 @@ func (s *Server) sqlCheckController(c echo.Context) error {
 			Name:  metricAPI.SQLAdviseAPIMetricName,
 			Value: 1,
 			Labels: map[string]string{
-				"database_type": dbType,
+				"database_type": string(dbType),
 				"environment":   envName,
 			},
 		})
 	}
 
 	return c.JSON(http.StatusOK, adviceList)
 }
+
+func (s *Server) findDatabase(ctx context.Context, host string, port string, databaseName string) (*api.Database, error) {
+	instanceList, err := s.store.FindInstance(ctx, &api.InstanceFind{
+		Host: &host,
+		Port: &port,
+	})
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to find instance by host: %v, port: %v", host, port)).SetInternal(err)
+	}
+	if len(instanceList) == 0 {
+		return nil, echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Cannot find instance with host: %v, port: %v", host, port))
+	}
+
+	var instanceIDList []int
+	for _, instance := range instanceList {
```

Comment:
All the instances should be the same. We can just pick the first one so that we don't need to change the find API.

Reply:
Updated